finat/sympy2gem.py

```diff
@@ -65,3 +65,72 @@ def sympy2gem_symbol(node, self):
 @sympy2gem.register(symengine.Rational)
 def sympy2gem_rational(node, self):
     return gem.Division(*(map(self, node.as_numer_denom())))
+
+
+@sympy2gem.register(sympy.Abs)
+@sympy2gem.register(symengine.Abs)
+def sympy2gem_abs(node, self):
+    return gem.MathFunction("abs", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Not)
+@sympy2gem.register(symengine.Not)
+def sympy2gem_not(node, self):
+    return gem.LogicalNot(*map(self, node.args))
+
+
+@sympy2gem.register(sympy.Or)
+@sympy2gem.register(symengine.Or)
+def sympy2gem_or(node, self):
+    return reduce(gem.LogicalOr, map(self, node.args))
+
+
+@sympy2gem.register(sympy.And)
+@sympy2gem.register(symengine.And)
+def sympy2gem_and(node, self):
+    return reduce(gem.LogicalAnd, map(self, node.args))
+
+
+@sympy2gem.register(sympy.Eq)
+@sympy2gem.register(symengine.Eq)
+def sympy2gem_eq(node, self):
+    return gem.Comparison("==", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Gt)
+def sympy2gem_gt(node, self):
+    return gem.Comparison(">", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Ge)
+def sympy2gem_ge(node, self):
+    return gem.Comparison(">=", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Lt)
+@sympy2gem.register(symengine.Lt)
+def sympy2gem_lt(node, self):
+    return gem.Comparison("<", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Le)
+@sympy2gem.register(symengine.Le)
+def sympy2gem_le(node, self):
+    return gem.Comparison("<=", *map(self, node.args))
+
+
+@sympy2gem.register(sympy.Piecewise)
+@sympy2gem.register(symengine.Piecewise)
+def sympy2gem_conditional(node, self):
+    expr = None
+    pieces = []
+    for v, c in node.args:
+        if c == True:  # noqa: E712
+            expr = self(v)
+            break
+        pieces.append((v, c))
+    if expr is None:
+        expr = gem.Literal(float("nan"))
+    for v, c in reversed(pieces):
```

Comment:
@wence- are you sure about reversing the order here?

Reply:
Pretty sure, you want the last condition to be made "first", because you keep wrapping the new conditions outside.

Consider:

```
(c1, v1), (c2, v2), (c3, v3), (True, v4)
```

I think that wants to be turned into:

```
Conditional(c1, v1, Conditional(c2, v2, Conditional(c3, v3, v4)))
```

So if you go from the back to the front, you build up the expressions:

```
expr = v4
expr = Conditional(c3, v3, expr) # v4
expr = Conditional(c2, v2, expr) # 
expr = Conditional(c1, v1, expr)
```
Which is what you want, no?